test_provider.py

```diff
@@ -0,0 +1,310 @@
+        #print(json['timeAccountAllocationCategories'])
+        ta = TimeAllocation(TimeAccountingCode(json['timeAccountAllocationCategories'][0]['category']),
+                            timedelta(milliseconds=json['awardedTime']),
+                            timedelta(milliseconds=0),
+                            timedelta(milliseconds=json['timeAccountAllocationCategories'][0]['programTime']),
+                            timedelta(milliseconds=json['timeAccountAllocationCategories'][0]['partnerTime']))
+        return ta
+
+    @staticmethod
+    def parse_or_group(json: dict) -> OrGroup:
+        
+        # Find nested OR groups/AND groups
+        # TODO: is this correct if there are not nested groups in OCS natively
+
+        observations = [JsonProvider.parse_observation(json[key], key) for key in json.keys() if key.startswith('OBSERVATION_BASIC')]
+
+        number_to_observe = len(observations)
+        delay_max, delay_min = 0, 0 # TODO: What are these?
+        or_group = OrGroup(json['key'], json['name'], number_to_observe, delay_min, delay_max, observations)
+        return or_group
+
+    @staticmethod
+    def parse_guide_star(json: dict) -> Resource:
+
+        # TODO: Maybe a GuideStart class should be in place to handle this better
+        res = Resource(json['name'], json['tag'])
+        return res
+    
+    @staticmethod
+    def parse_root_group(json: dict) -> OrGroup:
+        # Find nested OR groups/AND groups
+        groups = [JsonProvider.parse_and_group(json[key]) for key in json.keys() if key.startswith('GROUP_GROUP_SCHEDULING')]
+        num_to_observe = len(groups)
+        root_group = AndGroup(None, None, num_to_observe, 0, 0, groups, AndOption.ANYORDER)
+        return root_group
+
+    @staticmethod
+    def parse_and_group(json: dict) -> AndGroup:
+        observations = [JsonProvider.parse_observation(json[key], key) for key in json.keys() if key.startswith('OBSERVATION_BASIC')]
+
+        number_to_observe = len(observations)
+        delay_max, delay_min = 0, 0 # TODO: What are these?
+        # or_group = AndGroup(json['key'], json['name'], number_to_observe, delay_min, delay_max, observations, AndOption.ANYORDER)
+        return AndGroup(json['key'], json['name'], number_to_observe, delay_min, delay_max, observations, AndOption.ANYORDER)
+
+    @staticmethod
+    def parse_program(json: dict) -> Program:
+
+        ta = JsonProvider.parse_time_allocation(json)
+        root_group = JsonProvider.parse_root_group(json)
+        program = Program(json['programId'],
+                          json['key'],
+                          Band(int(json['queueBand'])),
+                          bool(json['isThesis']),
+                          ProgramMode(json['programMode']),
+                          None,
+                          None,
+                          ta,
+                          root_group,
+                          TooType(json['tooType']) if json['tooType'] != 'None' else None)
```

Comment:
I would really like us to stop relying on hard-coded string literals spread all over the code for things like this.

Let's try to keep them grouped to `Enum`s so that it's obvious where they are defined.

```
class ProgramLookups(Enum):
    ID = 'programId'
    KEY = 'key'
    BAND = 'queueBand'
....
json[ProgramLookups.ID]
....
```
Also, does the band come in as a string? Do we need to explicitly convert it to an `int`?